internal/handler/image.go

```diff
@@ -40,7 +40,7 @@ func ImageHandler(deps *service.Dependencies) http.HandlerFunc {
 
 		params := make(map[string]string)
 		values := r.URL.Query()
-		if len(values) > 0 {
+		if len(values) > 0 || len(deps.DefaultParams) > 0 {
```

Comment:
The `Manipulator` already has the information of default parameters. Wouldn't it make more sense to delegate the logic to `Manipulator` itself? It can return early, with no processing on image data, if default parameters are not present and the SpecBuilder also has not specified any params.

Reply:
We can. But from I understood from the code, the manipulator is built as an optional stage & gets triggered only when there is a manipulation param present. If you prefer to delegate this responsibility to the manipulator, it is fine too. It just makes the manipulator layer as a mandatory step.

Let me know the preference here, I will make the change, if needed.

Reply:
My concern here is injecting `DefaultParams` to `Dependencies`. If we can do something like
```go
if len(values) > 0 || deps.Manipulator.HasDefaultParams() {
    ...
}
```
This will be better IMO.

Reply:
Eventually we want to move this check to `URLParser`, if you can wait for that, we can avoid adding to the Manipulator interface contract.

Reply:
>My concern here is injecting DefaultParams to Dependencies. If we can do something like
Sure. Will make this change.